ibm/service/power/resource_ibm_pi_volume_clone.go

```diff
@@ -197,39 +217,17 @@ func isWaitForIBMPIVolumeCloneCompletion(ctx context.Context, client *st.IBMPICl
 	return stateConf.WaitForStateContext(ctx)
 }
 
-func isIBMPIVolumeCloneRefreshFunc(client *st.IBMPICloneVolumeClient, id string) resource.StateRefreshFunc {
+func isIBMPIVolumeCloneRefreshFunc(client *instance.IBMPICloneVolumeClient, id string) retry.StateRefreshFunc {
 	return func() (interface{}, string, error) {
 		volClone, err := client.Get(id)
 		if err != nil {
 			return nil, "", err
 		}
 
-		if *volClone.Status == VolumeCloneCompleted {
-			return volClone, VolumeCloneCompleted, nil
+		if *volClone.Status == State_Completed {
+			return volClone, State_Completed, nil
 		}
 
-		return volClone, VolumeCloneRunning, nil
-	}
-}
-
-func clonedVolumesSchema() *schema.Schema {
-	return &schema.Schema{
```

Comment:
This can still be used to removed repeating code.

Reply:
I'm not a fan of these loose functions that just return part of the schema. I think it makes it a bit harder to read. It does incur a bit of repetition, but I also don't know how/why which parts of the schema from the different resources/dats sources got chosen to be single functions.